### ISSUE TITLE
Installing qt@5 faild

### ISSUE THREAD

**Issue by byxrolland** [Homebrew/linuxbrew-core#23293]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
HOMEBREW_VERSION: 3.1.8-66-ga3e6a84
ORIGIN: https://github.com/Homebrew/brew.git
HEAD: a3e6a842ce6b744a38b493f85ff1ec58cd04a877
Last commit: 26 hours ago
Core tap ORIGIN: https://github.com/Homebrew/linuxbrew-core.git
Core tap HEAD: 80c98ec93dfea8a2901aba0342842af0c9a5f982
Core tap last commit: 3 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: 172.26.48.1:0.0
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/bin/ruby
CPU: octa-core 64-bit ivybridge
Clang: 12.0.0 build (parse error)
Git: 2.25.1 => /bin/git
Curl: 7.68.0 => /usr/bin/curl
Kernel: Linux 4.19.104-microsoft-standard x86_64 GNU/Linux
OS: Ubuntu 20.04.2 LTS (focal)
Host glibc: 2.31
/usr/bin/gcc: 9.3.0
/usr/bin/ruby: 2.7.0
glibc: N/A
gcc@5: N/A
xorg: N/A

Warning: Use libpthread-stubs instead of deprecated linuxbrew/xorg/libpthread-stubs
Warning: Use xorgproto instead of deprecated linuxbrew/xorg/xorgproto
Warning: Use libxau instead of deprecated linuxbrew/xorg/libxau
Warning: Use libx11 instead of deprecated linuxbrew/xorg/libx11
Warning: Use libxext instead of deprecated linuxbrew/xorg/libxext
Warning: Use libxfixes instead of deprecated linuxbrew/xorg/libxfixes
Warning: Use libpthread-stubs instead of deprecated linuxbrew/xorg/libpthread-stubs
Warning: Use xorgproto instead of deprecated linuxbrew/xorg/xorgproto
Warning: Use libxau instead of deprecated linuxbrew/xorg/libxau
Warning: Use libx11 instead of deprecated linuxbrew/xorg/libx11
Warning: Use libpthread-stubs instead of deprecated linuxbrew/xorg/libpthread-stubs
Warning: Use xorgproto instead of deprecated linuxbrew/xorg/xorgproto
Warning: Use libxau instead of deprecated linuxbrew/xorg/libxau
Warning: Use libx11 instead of deprecated linuxbrew/xorg/libx11
Warning: Use libxext instead of deprecated linuxbrew/xorg/libxext
Warning: Use libpthread-stubs instead of deprecated linuxbrew/xorg/libpthread-stubs
Warning: Use xorgproto instead of deprecated linuxbrew/xorg/xorgproto
Warning: Use libxau instead of deprecated linuxbrew/xorg/libxau
Warning: Use libx11 instead of deprecated linuxbrew/xorg/libx11
Warning: Use libxfixes instead of deprecated linuxbrew/xorg/libxfixes
Warning: Use xorgproto instead of deprecated linuxbrew/xorg/xorgproto
Warning: Use libpthread-stubs instead of deprecated linuxbrew/xorg/libpthread-stubs
Warning: Use xorgproto instead of deprecated linuxbrew/xorg/xorgproto
Warning: Use libxau instead of deprecated linuxbrew/xorg/libxau
Warning: Use libx11 instead of deprecated linuxbrew/xorg/libx11
Warning: Use libxext instead of deprecated linuxbrew/xorg/libxext
Warning: Use xorgproto instead of deprecated linuxbrew/xorg/xorgproto
Warning: Use libfontenc instead of deprecated linuxbrew/xorg/libfontenc
Warning: Use xorgproto instead of deprecated linuxbrew/xorg/xorgproto
Warning: Use libpthread-stubs instead of deprecated linuxbrew/xorg/libpthread-stubs
Warning: Use xorgproto instead of deprecated linuxbrew/xorg/xorgproto
Warning: Use libxau instead of deprecated linuxbrew/xorg/libxau
Warning: Use libx11 instead of deprecated linuxbrew/xorg/libx11
Warning: Use libxext instead of deprecated linuxbrew/xorg/libxext
Warning: Use libpthread-stubs instead of deprecated linuxbrew/xorg/libpthread-stubs
Warning: Use xorgproto instead of deprecated linuxbrew/xorg/xorgproto
Warning: Use libxau instead of deprecated linuxbrew/xorg/libxau
Warning: Use libx11 instead of deprecated linuxbrew/xorg/libx11
Warning: Use libxfixes instead of deprecated linuxbrew/xorg/libxfixes
Warning: Use libpthread-stubs instead of deprecated linuxbrew/xorg/libpthread-stubs
Warning: Use xorgproto instead of deprecated linuxbrew/xorg/xorgproto
Warning: Use libxau instead of deprecated linuxbrew/xorg/libxau
Warning: Use libx11 instead of deprecated linuxbrew/xorg/libx11
Warning: Use libxext instead of deprecated linuxbrew/xorg/libxext
Warning: Use libpthread-stubs instead of deprecated linuxbrew/xorg/libpthread-stubs
Warning: Use xorgproto instead of deprecated linuxbrew/xorg/xorgproto
Warning: Use libxau instead of deprecated linuxbrew/xorg/libxau
Warning: Use libx11 instead of deprecated linuxbrew/xorg/libx11
Warning: Use libxext instead of deprecated linuxbrew/xorg/libxext
Warning: Use libpthread-stubs instead of deprecated linuxbrew/xorg/libpthread-stubs
Warning: Use xorgproto instead of deprecated linuxbrew/xorg/xorgproto
Warning: Use libxau instead of deprecated linuxbrew/xorg/libxau
Warning: Use libx11 instead of deprecated linuxbrew/xorg/libx11
Warning: Use libpthread-stubs instead of deprecated linuxbrew/xorg/libpthread-stubs
Warning: Use xorgproto instead of deprecated linuxbrew/xorg/xorgproto
Warning: Use libxau instead of deprecated linuxbrew/xorg/libxau
Warning: Use libx11 instead of deprecated linuxbrew/xorg/libx11
Warning: Use libxfixes instead of deprecated linuxbrew/xorg/libxfixes
Warning: Use libpthread-stubs instead of deprecated linuxbrew/xorg/libpthread-stubs
Warning: Use xorgproto instead of deprecated linuxbrew/xorg/xorgproto
Warning: Use libxau instead of deprecated linuxbrew/xorg/libxau
Warning: Use libx11 instead of deprecated linuxbrew/xorg/libx11
Warning: Use libxext instead of deprecated linuxbrew/xorg/libxext
Warning: Use libice instead of deprecated linuxbrew/xorg/libice
Warning: Use libsm instead of deprecated linuxbrew/xorg/libsm
Warning: Use libxt instead of deprecated linuxbrew/xorg/libxt
Warning: Use libxmu instead of deprecated linuxbrew/xorg/libxmu
Warning: Use libxpm instead of deprecated linuxbrew/xorg/libxpm
Warning: Use libpthread-stubs instead of deprecated linuxbrew/xorg/libpthread-stubs
Warning: Use xorgproto instead of deprecated linuxbrew/xorg/xorgproto
Warning: Use libxau instead of deprecated linuxbrew/xorg/libxau
Warning: Use libx11 instead of deprecated linuxbrew/xorg/libx11
Warning: Use libxext instead of deprecated linuxbrew/xorg/libxext
Your system is ready to brew.
```

### 

- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.

### What were you trying to do (and why)?

Install qt@5.

### What happened (include all command output)?

```
g++ -c -o qlogging.o   -std=c++11 -ffunction-sections -fdata-sections -g -g  -I/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/qmake -I/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/qmake/library -I/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/qmake/generators -I/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/src/3rdparty/tinycbor/src -I/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/qmake/generators/unix -I/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/qmake/generators/win32 -I/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/qmake/generators/mac -I/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include -I/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore -I/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/5.15.2 -I/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/5.15.2/QtCore -I../src/corelib/global -I/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/mkspecs/linux-g++ -DQT_VERSION_STR=\"5.15.2\" -DQT_VERSION_MAJOR=5 -DQT_VERSION_MINOR=15 -DQT_VERSION_PATCH=2 -DQT_BUILD_QMAKE -DQT_BOOTSTRAPPED -DPROEVALUATOR_FULL -DQT_NO_FOREACH /tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/src/corelib/global/qlogging.cpp
In file included from /tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/qfloat16.h:1,
                 from /tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/src/corelib/global/qendian.h:44,
                 from /tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/src/corelib/global/qendian.cpp:41:
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qfloat16.h:300:7: error: 'numeric_limits' is not a class template
  300 | class numeric_limits<QT_PREPEND_NAMESPACE(qfloat16)> : public numeric_limits<float>
      |       ^~~~~~~~~~~~~~
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qfloat16.h:300:77: error: expected template-name before '<' token
  300 | class numeric_limits<QT_PREPEND_NAMESPACE(qfloat16)> : public numeric_limits<float>
      |                                                                             ^
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qfloat16.h:300:77: error: expected '{' before '<' token
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qfloat16.h:344:18: error: 'numeric_limits' is not a class template
  344 | template<> class numeric_limits<const QT_PREPEND_NAMESPACE(qfloat16)>
      |                  ^~~~~~~~~~~~~~
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qfloat16.h:344:69: error: 'std::numeric_limits' is not a template
  344 | template<> class numeric_limits<const QT_PREPEND_NAMESPACE(qfloat16)>
      |                                                                     ^
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qfloat16.h:300:7: note: previous declaration here
  300 | class numeric_limits<QT_PREPEND_NAMESPACE(qfloat16)> : public numeric_limits<float>
      |       ^~~~~~~~~~~~~~
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qfloat16.h:345:28: error: expected template-name before '<' token
  345 |     : public numeric_limits<QT_PREPEND_NAMESPACE(qfloat16)> {};
      |                            ^
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qfloat16.h:346:18: error: 'numeric_limits' is not a class template
  346 | template<> class numeric_limits<volatile QT_PREPEND_NAMESPACE(qfloat16)>
      |                  ^~~~~~~~~~~~~~
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qfloat16.h:346:72: error: 'std::numeric_limits' is not a template
  346 | template<> class numeric_limits<volatile QT_PREPEND_NAMESPACE(qfloat16)>
      |                                                                        ^
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qfloat16.h:300:7: note: previous declaration here
  300 | class numeric_limits<QT_PREPEND_NAMESPACE(qfloat16)> : public numeric_limits<float>
      |       ^~~~~~~~~~~~~~
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qfloat16.h:347:28: error: expected template-name before '<' token
  347 |     : public numeric_limits<QT_PREPEND_NAMESPACE(qfloat16)> {};
      |                            ^
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qfloat16.h:348:18: error: 'numeric_limits' is not a class template
  348 | template<> class numeric_limits<const volatile QT_PREPEND_NAMESPACE(qfloat16)>
      |                  ^~~~~~~~~~~~~~
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qfloat16.h:348:78: error: 'std::numeric_limits' is not a template
  348 | template<> class numeric_limits<const volatile QT_PREPEND_NAMESPACE(qfloat16)>
      |                                                                              ^
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qfloat16.h:300:7: note: previous declaration here
  300 | class numeric_limits<QT_PREPEND_NAMESPACE(qfloat16)> : public numeric_limits<float>
      |       ^~~~~~~~~~~~~~
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qfloat16.h:349:28: error: expected template-name before '<' token
  349 |     : public numeric_limits<QT_PREPEND_NAMESPACE(qfloat16)> {};
      |                            ^
In file included from /tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/src/corelib/global/qendian.cpp:41:
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/src/corelib/global/qendian.h: In static member function 'static constexpr QSpecialInteger<S> QSpecialInteger<S>::max()':
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/src/corelib/global/qendian.h:331:30: error: 'std::numeric_limits' is not a template
  331 |     { return QSpecialInteger(std::numeric_limits<T>::max()); }
      |                              ^~~
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/src/corelib/global/qendian.h:331:54: error: incomplete type 'std::numeric_limits' used in nested name specifier
  331 |     { return QSpecialInteger(std::numeric_limits<T>::max()); }
      |                                                      ^~~
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/src/corelib/global/qendian.h: In static member function 'static constexpr QSpecialInteger<S> QSpecialInteger<S>::min()':
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/src/corelib/global/qendian.h:333:30: error: 'std::numeric_limits' is not a template
  333 |     { return QSpecialInteger(std::numeric_limits<T>::min()); }
      |                              ^~~
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/src/corelib/global/qendian.h:333:54: error: incomplete type 'std::numeric_limits' used in nested name specifier
  333 |     { return QSpecialInteger(std::numeric_limits<T>::min()); }
      |                                                      ^~~
In file included from /tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/qfloat16.h:1,
                 from /tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qendian.h:44,
                 from /tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/qendian.h:1,
                 from /tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/src/corelib/codecs/qutfcodec.cpp:43:
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qfloat16.h:300:7: error: 'numeric_limits' is not a class template
  300 | class numeric_limits<QT_PREPEND_NAMESPACE(qfloat16)> : public numeric_limits<float>
      |       ^~~~~~~~~~~~~~
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qfloat16.h:300:77: error: expected template-name before '<' token
  300 | class numeric_limits<QT_PREPEND_NAMESPACE(qfloat16)> : public numeric_limits<float>
      |                                                                             ^
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qfloat16.h:300:77: error: expected '{' before '<' token
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qfloat16.h:344:18: error: 'numeric_limits' is not a class template
  344 | template<> class numeric_limits<const QT_PREPEND_NAMESPACE(qfloat16)>
      |                  ^~~~~~~~~~~~~~
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qfloat16.h:344:69: error: 'std::numeric_limits' is not a template
  344 | template<> class numeric_limits<const QT_PREPEND_NAMESPACE(qfloat16)>
      |                                                                     ^
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qfloat16.h:300:7: note: previous declaration here
  300 | class numeric_limits<QT_PREPEND_NAMESPACE(qfloat16)> : public numeric_limits<float>
      |       ^~~~~~~~~~~~~~
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qfloat16.h:345:28: error: expected template-name before '<' token
  345 |     : public numeric_limits<QT_PREPEND_NAMESPACE(qfloat16)> {};
      |                            ^
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qfloat16.h:346:18: error: 'numeric_limits' is not a class template
  346 | template<> class numeric_limits<volatile QT_PREPEND_NAMESPACE(qfloat16)>
      |                  ^~~~~~~~~~~~~~
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qfloat16.h:346:72: error: 'std::numeric_limits' is not a template
  346 | template<> class numeric_limits<volatile QT_PREPEND_NAMESPACE(qfloat16)>
      |                                                                        ^
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qfloat16.h:300:7: note: previous declaration here
  300 | class numeric_limits<QT_PREPEND_NAMESPACE(qfloat16)> : public numeric_limits<float>
      |       ^~~~~~~~~~~~~~
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qfloat16.h:347:28: error: expected template-name before '<' token
  347 |     : public numeric_limits<QT_PREPEND_NAMESPACE(qfloat16)> {};
      |                            ^
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qfloat16.h:348:18: error: 'numeric_limits' is not a class template
  348 | template<> class numeric_limits<const volatile QT_PREPEND_NAMESPACE(qfloat16)>
      |                  ^~~~~~~~~~~~~~
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qfloat16.h:348:78: error: 'std::numeric_limits' is not a template
  348 | template<> class numeric_limits<const volatile QT_PREPEND_NAMESPACE(qfloat16)>
      |                                                                              ^
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qfloat16.h:300:7: note: previous declaration here
  300 | class numeric_limits<QT_PREPEND_NAMESPACE(qfloat16)> : public numeric_limits<float>
      |       ^~~~~~~~~~~~~~
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qfloat16.h:349:28: error: expected template-name before '<' token
  349 |     : public numeric_limits<QT_PREPEND_NAMESPACE(qfloat16)> {};
      |                            ^
In file included from /tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/qendian.h:1,
                 from /tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/src/corelib/codecs/qutfcodec.cpp:43:
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qendian.h: In static member function 'static constexpr QSpecialInteger<S> QSpecialInteger<S>::max()':
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qendian.h:331:30: error: 'std::numeric_limits' is not a template
  331 |     { return QSpecialInteger(std::numeric_limits<T>::max()); }
      |                              ^~~
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qendian.h:331:54: error: incomplete type 'std::numeric_limits' used in nested name specifier
  331 |     { return QSpecialInteger(std::numeric_limits<T>::max()); }
      |                                                      ^~~
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qendian.h: In static member function 'static constexpr QSpecialInteger<S> QSpecialInteger<S>::min()':
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qendian.h:333:30: error: 'std::numeric_limits' is not a template
  333 |     { return QSpecialInteger(std::numeric_limits<T>::min()); }
      |                              ^~~
/tmp/qtA5-20210523-3687-8wqjdv/qt-everywhere-src-5.15.2/qtbase/include/QtCore/../../src/corelib/global/qendian.h:333:54: error: incomplete type 'std::numeric_limits' used in nested name specifier
  333 |     { return QSpecialInteger(std::numeric_limits<T>::min()); }
      |                                                      ^~~
make: *** [Makefile:383: qutfcodec.o] Error 1
make: *** Waiting for unfinished jobs....
make: *** [Makefile:341: qendian.o] Error 1

==> Formula
Path: /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/qt@5.rb
==> Configuration
HOMEBREW_VERSION: 3.1.8-66-ga3e6a84
ORIGIN: https://github.com/Homebrew/brew.git
HEAD: a3e6a842ce6b744a38b493f85ff1ec58cd04a877
Last commit: 26 hours ago
Core tap ORIGIN: https://github.com/Homebrew/linuxbrew-core.git
Core tap HEAD: 80c98ec93dfea8a2901aba0342842af0c9a5f982
Core tap last commit: 3 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: 172.26.48.1:0.0
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/bin/ruby
CPU: octa-core 64-bit ivybridge
Clang: 12.0.0 build (parse error)
Git: 2.25.1 => /bin/git
Curl: 7.68.0 => /usr/bin/curl
Kernel: Linux 4.19.104-microsoft-standard x86_64 GNU/Linux
OS: Ubuntu 20.04.2 LTS (focal)
Host glibc: 2.31
/usr/bin/gcc: 9.3.0
/usr/bin/ruby: 2.7.0
glibc: N/A
gcc@5: N/A
xorg: N/A
==> ENV
HOMEBREW_CC: gcc-11
HOMEBREW_CXX: g++-11
MAKEFLAGS: -j8
CMAKE_PREFIX_PATH: /home/linuxbrew/.linuxbrew/opt/python@3.8:/home/linuxbrew/.linuxbrew
PKG_CONFIG_PATH: /home/linuxbrew/.linuxbrew/opt/ncurses/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/readline/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/zlib/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libxml2/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/sqlite/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/expat/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/dbus/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libffi/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/pcre/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/openssl@1.1/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/xz/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/python@3.9/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/util-linux/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/glib/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libpthread-stubs/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libxau/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libxdmcp/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libxcb/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libx11/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libxext/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libxfixes/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libxi/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libxtst/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/at-spi2-core/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libpng/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/freetype/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/fontconfig/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/icu4c/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libproxy/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libxkbcommon/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libice/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libsm/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libxcomposite/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libpciaccess/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libdrm/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libxdamage/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libedit/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libelf/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/gmp/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/isl/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/mpfr/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libxrender/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libxrandr/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libxshmfence/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libxv/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libxvmc/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libxxf86vm/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/wayland/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libva/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libvdpau/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/mesa/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/json-c/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libogg/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/flac/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libvorbis/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/opus/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libsndfile/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libsoxr/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/speexdsp/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libcap/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/pulseaudio/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/tcl-tk/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/python@3.8/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libxcursor/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libxscrnsaver/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/libxinerama/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/sdl2/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/lz4/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/systemd/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/xcb-util/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/xcb-util-image/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/xcb-util-keysyms/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/xcb-util-renderutil/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/xcb-util-wm/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/zstd/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/xorgproto/share/pkgconfig:/home/linuxbrew/.linuxbrew/opt/xkeyboardconfig/share/pkgconfig:/home/linuxbrew/.linuxbrew/opt/wayland-protocols/share/pkgconfig:/home/linuxbrew/.linuxbrew/opt/systemd/share/pkgconfig
HOMEBREW_GIT: git
ACLOCAL_PATH: /home/linuxbrew/.linuxbrew/share/aclocal
PATH: /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/shims/linux/super:/home/linuxbrew/.linuxbrew/opt/pkg-config/bin:/home/linuxbrew/.linuxbrew/opt/gperf/bin:/home/linuxbrew/.linuxbrew/opt/m4/bin:/home/linuxbrew/.linuxbrew/opt/bison/bin:/home/linuxbrew/.linuxbrew/opt/ncurses/bin:/home/linuxbrew/.linuxbrew/opt/libxml2/bin:/home/linuxbrew/.linuxbrew/opt/gettext/bin:/home/linuxbrew/.linuxbrew/opt/flex/bin:/home/linuxbrew/.linuxbrew/opt/sqlite/bin:/home/linuxbrew/.linuxbrew/opt/expat/bin:/home/linuxbrew/.linuxbrew/opt/dbus/bin:/home/linuxbrew/.linuxbrew/opt/bzip2/bin:/home/linuxbrew/.linuxbrew/opt/pcre/bin:/home/linuxbrew/.linuxbrew/opt/gdbm/bin:/home/linuxbrew/.linuxbrew/opt/openssl@1.1/bin:/home/linuxbrew/.linuxbrew/opt/xz/bin:/home/linuxbrew/.linuxbrew/opt/unzip/bin:/home/linuxbrew/.linuxbrew/opt/python@3.9/bin:/home/linuxbrew/.linuxbrew/opt/util-linux/bin:/home/linuxbrew/.linuxbrew/opt/glib/bin:/home/linuxbrew/.linuxbrew/opt/libpng/bin:/home/linuxbrew/.linuxbrew/opt/freetype/bin:/home/linuxbrew/.linuxbrew/opt/fontconfig/bin:/home/linuxbrew/.linuxbrew/opt/icu4c/bin:/home/linuxbrew/.linuxbrew/opt/libproxy/bin:/home/linuxbrew/.linuxbrew/opt/libxkbcommon/bin:/home/linuxbrew/.linuxbrew/opt/binutils/bin:/home/linuxbrew/.linuxbrew/opt/gcc/bin:/home/linuxbrew/.linuxbrew/opt/llvm/bin:/home/linuxbrew/.linuxbrew/opt/lm-sensors/bin:/home/linuxbrew/.linuxbrew/opt/wayland/bin:/home/linuxbrew/.linuxbrew/opt/mesa/bin:/home/linuxbrew/.linuxbrew/opt/flac/bin:/home/linuxbrew/.linuxbrew/opt/libsndfile/bin:/home/linuxbrew/.linuxbrew/opt/libtool/bin:/home/linuxbrew/.linuxbrew/opt/pulseaudio/bin:/home/linuxbrew/.linuxbrew/opt/tcl-tk/bin:/home/linuxbrew/.linuxbrew/opt/python@3.8/bin:/home/linuxbrew/.linuxbrew/opt/xinput/bin:/home/linuxbrew/.linuxbrew/opt/sdl2/bin:/home/linuxbrew/.linuxbrew/opt/lz4/bin:/home/linuxbrew/.linuxbrew/opt/systemd/bin:/home/linuxbrew/.linuxbrew/opt/zstd/bin:/usr/bin:/bin:/usr/sbin:/sbin

Error: qt@5 5.15.2 did not build
```

### What did you expect to happen?

I expect the installation to complete.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
brew install qt@5
```
```

**Comment by hktonylee** [comment]
Same for me. Here is the log - https://gist.github.com/67057d4396299e716db1c2a73b4d1e40

**Comment by danielnachun** [comment]
Here is where we have been working on trying to fix this: https://github.com/Homebrew/linuxbrew-core/pull/23174.  I haven't had a chance to try to push more fixes to this recently, but the main issue is just getting to build with the right GCC.  This is normally a relocatable bottle that nobody should need to compile, but some recent changes with LLVM have created some new build difficulties.

**Comment by mustiboost** [comment]
I have the exact same issue. I hope this gets resolved soon as other packages that depend on qt5 won't install either. One such package is OpenCV.  Is there a temporary workaround?

**Comment by danielnachun** [comment]
The only workaround would be to disable Qt support in the formula which depend on it.  In the case of OpenCV, it depends on Qt through VTK.  So disabling Qt support in VTK would let you remove Qt as a dependency for that formula, and consequently OpenCV wouldn't depend on it either.  I think the Gentoo ebuild is a good starting point for how to disable Qt in VTK: https://gitweb.gentoo.org/repo/gentoo.git/tree/sci-libs/vtk/vtk-8.2.0.ebuild.  

This is definitely something we want to have fixed (along with Qt 6, also currently broken).  But Qt is probably one of the most complicated formulae we ship, so it often takes a while to figure out what to fix.

**Comment by mustiboost** [comment]
Got it, thanks! Your efforts are appreciated! :+1:

**Comment by jasjuang** [comment]
Running into the same problem and looking forward to a fix

**Comment by polson** [comment]
I also have this problem using WSL on Windows

**Comment by galderaa** [comment]
Same problem on Ubuntu 20.04.
is there a way to revert the brew to the time that it was working?

**Comment by satmandu** [comment]
I think the best bet for Qt5 going forward is going to be using the kde patched 5.15 at https://invent.kde.org/qt/qt/qtbase/-/tree/kde/5.15

**Comment by danielnachun** [comment]
We will be able to build this in homebrew-core soon, and once we do it should also be more likely to build from source.  The limitation at the moment is that Github Actions VM does not have a enough disk space to fit all the dependencies, but in homebrew-core we have a self-hosted Google Cloud VM without this limitation.  We need to fix an issue that is blocking us from building some dependencies of `qt@5`, and then we should have it build soon.  

For anyone feeling adventurous, you can try to use the changes I added in this pull request: https://github.com/Homebrew/linuxbrew-core/pull/23174/files.  I could not get it to build from source locally, but I haven't tried recently.  I had to add some upstream patches to fix things for `gcc-11` but others may have success.

**Comment by satmandu** [comment]
FYI, I got this to compile for Chromebrew last night for architectures `x86_64` and `armv7l`. Here's our PR without any special patches, just using the KDE tree: https://github.com/skycocker/chromebrew/pull/6054

**Comment by danielnachun** [comment]
I think we will end up taking commits from that KDE tree.  We try to avoid using untagged source trees whenever possible but hopefully we can figure out the right set of changes from the 5.15.2 release needed for it to build.

**Comment by iMichka** [comment]
Duplicate of #23174
